spec/libs/twitter_processor_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+RSpec.describe TwitterProcessor do
+  let!(:company) { create(:company, twitter: 'https://twitter.com/netflix') }
+  let!(:processor) do
+    TwitterProcessor.new(company: company, number_of_tweets: 6)
+  end
+
+  describe 'screen name' do
+    context 'with twitter link valid' do
+      it 'screen name should be correct' do
+        expect(processor.screen_name).to eq('netflix')
+      end
+    end
+
+    context 'with twitter link empty' do
+      it 'screen name should be empty' do
+        company.update_columns(twitter: nil)
+        processor = TwitterProcessor.new(company: company, number_of_tweets: 6)
+
+        expect(processor.screen_name).to be(nil)
+      end
+    end
+  end
+
+  describe 'tweets' do
+    context 'working scrape' do
+      let!(:response_fixture) { Rails.root.join('spec/fixtures/tweets.json').to_s }
+      let!(:response_json) { Nokogiri::HTML(File.read(response_fixture)) }
+      before(:each) do
+        allow(processor).to receive(:scrape_tweets).and_return(response_json)
```

Comment:
it's saying `response_json` but actually `Nokogiri::HTML`

Reply:
Here I ran in to problem i couldn't solve. 'JSON.parse' somehow doesn't work. The file i saved by hand scraping from twitter and doing `file << tweets.to_json`. with 'nokogiri' test works but doesn't with `json.parse`

Reply:
will give a try to `VCR`. hopefully it will give some results